src/vs/workbench/services/host/browser/host.ts

```diff
@@ -88,5 +88,11 @@ export interface IHostService {
 	 */
 	close(): Promise<void>;
 
+	/**
+	 * Execute an asynchronous `expectedShutdownTask`. While this task is
+	 * in progress, attempts to quit the application will not be vetoed with a dialog.
+	 */
+	withSilentShutdown?<T>(expectedShutdownTask: () => Promise<T>): Promise<T>;
```

Comment:
Naming is hard, any thoughts on something more expressive?

Reply:
How about `withExpectedShutdown`?

Reply:
Yeah!